stan/math/opencl/opencl_context.hpp

```diff
@@ -35,14 +35,12 @@ namespace stan {
 namespace math {
 namespace opencl {
 /**
- * A helper function to convert a std::vector
- * to cl::size_t<3>
+ * A helper function to convert an array to a cl::size_t<N>
  */
-template <size_t len = 3>
-cl::size_t<len> to_size_t(std::vector<size_t> values) {
-  assert(values.size() == len);
-  cl::size_t<len> s;
-  for (size_t i = 0; i < len; i++)
+template <int N>
+auto to_size_t(const size_t (&values)[N]) {
```

Comment:
Are there any downstream tests that should fail from this not asserting?

Is there some reasonable limit N should be? That’s be nice to doc since this is a template type. (Unless c++14 now goes with that being a const expr instead of what it used to do.... still catching up on changes to the lang)

Reply:
The way cl::size_t si defined now, N is always going to be 3. This cl::size_t is really weirdly defined and was a topic of a lot of discussions on the Khronos group dev boards. You can see why.

Reply:
Ah. I see. If that’s the intention, then one way that might make sense is to make the base template throw (and not have a meaningful implementation). And then have a template specialization for N = 3. That way you get the best of both worlds. 

There’s a potentially better solution, but I don’t know if it’s possible. For the base template, have it not be compilable. That’ll stop things at compile time. But once again, not sure if that’s easy to set up.

Reply:
And if you don’t want that, just adding that info to the function doc would be useful to anyone that comes across this in the future.